checkov/cloudformation/graph_builder/graph_components/blocks.py

```diff
@@ -66,9 +66,7 @@ def update_attribute(
                     key_to_update = ".".join(attribute_key_parts[i:])
                     break
 
-            if isinstance(obj_to_update, list):
-                key_to_update = int(key_to_update)
```

Comment:
if remember correctly this was needed for some specific use cases that's why the UTs are failing. Even I don't like, maybe we can add a try/except block. Do you have the problematic template file?